lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -1236,37 +1240,38 @@ function updateIntervalsFromIsoSegments(intervals, isoSegments, time, wmsItem) {
         // So it looks like Map Server defines a period in GetCapabilities, but actually wants it requested using a date,
         // not a period, and that any date in that interval will return the same thing.
         intervals.addInterval(new TimeInterval({
-            start: JulianDate.fromIso8601(start.format()),
-            stop: JulianDate.fromIso8601(stop.format()),
+            start: JulianDate.fromDate(start),
+            stop: JulianDate.fromDate(stop),
             data: start  // Convert the period to a date for requests (see discussion above).
         }));
     } else {
         // Note WMS uses extension ISO19128 of ISO8601; ISO 19128 allows start/end/periodicity
         // and does not use the "R[n]/" prefix for repeated intervals
         // eg. Data refreshed every 30 min: 2000-06-18T14:30Z/2000-06-18T14:30Z/PT30M
         // See 06-042_OpenGIS_Web_Map_Service_WMS_Implementation_Specification.pdf section D.4
-        var duration = moment.duration(isoSegments[2]);
-        if (duration.isValid() &&
-            (duration.milliseconds() > 0 || duration.seconds() > 0 || duration.minutes() > 0 ||
-             duration.hours() > 0 || duration.days() > 0 || duration.weeks() > 0 ||
-             duration.months() > 0 || duration.years() > 0)) {
-
-            var thisStop = start.clone();
+        var duration = isoDuration.parse(isoSegments[2]);
+        if (duration.seconds > 0 || duration.minutes > 0 ||
+            duration.hours > 0 || duration.days > 0 || duration.weeks > 0 ||
+            duration.months > 0 || duration.years > 0) {
+
+            var durationAsSeconds = isoDuration.toSeconds(duration);
```

Comment:
We can't just convert the duration to seconds, because years and months, at least, are different numbers of seconds. 1 month means 1 month, not 30 days.

Reply:
🤦 tried to be smart but failed!